src/Rules/Data.hs

```diff
@@ -28,6 +28,10 @@ buildPackageData context@Context {..} = do
         -- GhcCabal may run the configure script, so we depend on it.
         whenM (doesFileExist $ configure <.> "ac") $ need [configure]
 
+        -- XXX (izgzhen): rebuild @package-data.mk@ would need @include/settings@ etc.
```

Comment:
Sometimes the build system will complain `settings` is not found (not copied to `prefix/lib/include`) when building `package-data.mk` for some package. So I added this rule as a hot fix -- but this caused the recursive dependency as we saw earlier.

`XXX` might mean that I don't have a good idea how to deal with this, because when building from scratch we are okay without it.